File: A_LeetCode_Grinding_Guide/贪心算法/区间问题/Non_overlappingIntervals.java
import java.util.Arrays;
import java.util.Comparator;
public class Non_overlappingIntervals {
    public static int intervalSchedule(int[][] intervals) {
        if (intervals.length == 0) return 0;
        int n=intervals.length;
        // 按 b 升序排序
        Arrays.sort(intervals, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                return a[1] - b[1];
            }
        });
        int total=0;
        int prev=intervals[0][1];
        for (int i=1;i<n;i++){
            if(intervals[i][0]<prev){
                total+=1;
            }
            else{
                prev=intervals[i][1];
            }
        }
        return total;
    }
    public static void main(String[] args){
        int [][] intvs=new int[][]{{1, 2}};
        System.out.println(intervalSchedule(intvs));
    }
}

File: A_LeetCode_Grinding_Guide/贪心算法/区间问题/Non_overlappingIntervals.py
class Non_overlappingIntervals:
    def eraseOverlapIntervals(self, intervals: list) -> int:
        if not intervals:
            return 0
        n=len(intervals)
        intervals = sorted(intervals, key=lambda b:b[1])#按照数组中的[[a,b]]的b位从小到大排序
        total,prev=0,intervals[0][1]
        for i in range(1,n):
            if (intervals[i][0]<prev):
                total+=1
            else:
                prev=intervals[i][1]
        return total

result=Non_overlappingIntervals()
intervals=[[1,2]]
print(result.eraseOverlapIntervals(intervals))
